### PR TITLE
merge methods FlexboxHelper#calculateHorizontalFlexLines() and FlexboxHelper#calculateVerticalFlexLines()

### PR DESCRIPTION
**Pull request by Gavras**
calculateHorizontalFlexLines() and calculateVerticalFlexLines() have more than 100 lines.
They are mostly similar.

Tell me if you think it is a good idea to merge them.
I merged them into one method. I made helper methods.
There are a lot more similar methods like them you can merge
and basically refer to horizontal and vertical as one.

note: I didn't changed the logic behind the algorithm.

### CODE REVIEW

**Comment by Gavras** [comment]
I signed it!

**Comment by Gavras** [comment]
Do you mean OrientationHelper?

**Comment by thagikura** [comment]
Sorry, yes I meant OrientationHelper.

**Comment by Gavras** [comment]
As far as I know OrientationHelper uses RecyclerView.LayoutManager.
If there is a specific design I didn't consider I'd be happy to learn and implement it here.

What I did is I looked at FlexboxLayoutManager#getDecorationLengthMainAxis() and FlexboxLayoutManager#getDecorationLengthCrossAxis() and all the methods that use FlexboxLayoutManager#isMainAxisDirectionHorizontal() and I saw the consistency.

How would you recommend me to face OrientationHelper not working with ViewGroup?

**Comment by thagikura** [comment]
Hi @Gavras, 

Apologies for the belated reply.
After another consideration, it looks like it seems better not to use an OrientationHelper because it may be overkill.
So your current suggestion looks good overall.

But few suggestion, instead of adding isMainAxisHorizontal in the FlexboxHelper, could you add an method   to the FlexContainer interface and have two classes (FlexboxLayoutManager and FlexboxLayout) override the method? There should be already the same method in each of them.

Also I accidentally pushed the dev_recyclerview branch to incorporate the changes in master.
So could you apply your change on top of the current dev_recyclerview branch?

Thanks!

**Comment by Gavras** [comment]
OK I moved the method to the interface.

File: flexbox/src/main/java/com/google/android/flexbox/FlexboxHelper.java [hunk 1]
```diff
@@ -519,67 +329,82 @@ FlexLinesResult calculateVerticalFlexLines(int widthMeasureSpec,
...
     }
 
     /**
      * Calculate how many flex lines are needed in the flex container layout by measuring each
-     * child when the direction of the flex line is vertical (top to bottom or bottom to top).
+     * child.
      * Expand or shrink the flex items depending on the flex grow and flex shrink
-     * attributes in a later procedure, so views measured width may be changed in a later process
-     * or calculating the flex container.
+     * attributes in a later procedure, so views measured width\height may be changed
```
Comment (thagikura) [inline flexbox/src/main/java/com/google/android/flexbox/FlexboxHelper.java]: width/height ?

File: flexbox/src/main/java/com/google/android/flexbox/FlexboxHelper.java [hunk 2]
```diff
@@ -382,8 +384,6 @@ private FlexLinesResult calculateFlexLines(int mainMeasureSpec, int crossMeasure
 
         boolean reachedToIndex = toIndex == NO_POSITION;
 
-        boolean isMainHorizontal = mFlexContainer.isMainAxisDirectionHorizontal();
```
Comment (Gavras) [inline flexbox/src/main/java/com/google/android/flexbox/FlexboxHelper.java]: I thought the most important info in this method is if the main axis horizontal so I moved the variable to the start of the method.

**Comment by Gavras** [comment]
Sorry I forgot the test files.
And I've also changed the documentation where you commented, sorry for that too, it was saying width and as we are talking about the main axis I had to change it. Now I hope it's clearer.

**Comment by Gavras** [comment]
Sorry, I accidentally pushed the changes from another branch to here. I'll try to revert that and make another pull request after that.
The other branch was directing the issue with match_parent #196.
Also it deals with an issue that occurs while the first item in the line (not including the first line) has align item stretched and it becomes the size of the previous line instead of its own line.

**Comment by thagikura** [comment]
Ok no worries, thanks for the clarification.

**Comment by thagikura** [comment]
Ah also thanks for addressing the issue #196!!
Really appreciate it.

**Comment by Gavras** [comment]
I really sometimes don't get git. I seem to not found a way to revert just that as it was apparently a local branch and I afraid to lose the changes.
Can you look at it anyway?

Combining the methods is not included in the last commit,
I addressed what you asked me.

The last commit is about match_parent,
I think the flexline member name mIndicesAlignSelfStretch is not suitable anymore as it holds indices with match parent in their cross axis, it should be named to something like mIndicesToStretch.

Sorry about the inconvenient, I know it's better to break various issues into pieces.
And thanks.

**Comment by thagikura** [comment]
Hi @Gavras,

Apologies for the delay.

> I really sometimes don't get git. I seem to not found a way to revert just that as it was apparently a local 
> branch and I afraid to lose the changes.
> Can you look at it anyway?

Could you separate the two changes in different PRs?
Actually I'd like to merge each change separately so that it can be diffed easily and rolled back in case there is an issue.

If reverting the git commits is hard, you can send another PR for each change.
Thank you for your continuous effort! Really appreciate it.

**Comment by Gavras** [comment]
I send PR for calculateFlexLines() method.
If I got it right after the other PR this will remain with just the relevant changes to match_parent.

**Pull request closed without merging**